src/wavesurfer.js

```diff
@@ -284,9 +284,13 @@ var WaveSurfer = {
      * Directly load an externally decoded AudioBuffer.
      */
     loadDecodedBuffer: function (buffer) {
-        this.backend.load(buffer);
-        this.drawBuffer();
-        this.fireEvent('ready');
+        if (this.params.backend == 'Aurora') {
+            this.backend.load(buffer, this.drawAuroraBuffer.bind(this));
```

Comment:
I'd pass an anonymous function as the callback here, in which `drawAuroraBuffer` and `fireEvent('ready')` would be called.